lxd/db/projects.go

```diff
@@ -19,35 +19,43 @@ import (
-//go:generate mapper stmt -e project names-by-Name
-//go:generate mapper stmt -e project objects
-//go:generate mapper stmt -e project objects-by-Name
-//go:generate mapper stmt -e project used-by-ref
-//go:generate mapper stmt -e project used-by-ref-by-Name
-//go:generate mapper stmt -e project config-ref
-//go:generate mapper stmt -e project config-ref-by-Name
-//go:generate mapper stmt -e project create
-//go:generate mapper stmt -e project create-config-ref
-//go:generate mapper stmt -e project id
-//go:generate mapper stmt -e project rename
-//go:generate mapper stmt -e project update
-//go:generate mapper stmt -e project delete-by-Name
+//go:generate mapper stmt -p db -e project names
+//go:generate mapper stmt -p db -e project names-by-Name
+//go:generate mapper stmt -p db -e project objects
+//go:generate mapper stmt -p db -e project objects-by-Name
+//go:generate mapper stmt -p db -e project used-by-ref
+//go:generate mapper stmt -p db -e project used-by-ref-by-Name
+//go:generate mapper stmt -p db -e project config-ref
+//go:generate mapper stmt -p db -e project config-ref-by-Name
+//go:generate mapper stmt -p db -e project create struct=Project
+//go:generate mapper stmt -p db -e project create-config-ref
+//go:generate mapper stmt -p db -e project id
+//go:generate mapper stmt -p db -e project rename
+//go:generate mapper stmt -p db -e project update struct=Project
+//go:generate mapper stmt -p db -e project delete-by-Name
 //
-//go:generate mapper method -e project URIs
-//go:generate mapper method -e project List
-//go:generate mapper method -e project Get
-//go:generate mapper method -e project ConfigRef
-//go:generate mapper method -e project Exists
-//go:generate mapper method -e project Create
-//go:generate mapper method -e project UsedByRef
-//go:generate mapper method -e project ID
-//go:generate mapper method -e project Rename
-//go:generate mapper method -e project DeleteOne
+//go:generate mapper method -p db -e project URIs
+//go:generate mapper method -p db -e project List
+//go:generate mapper method -p db -e project Get struct=Project
+//go:generate mapper method -p db -e project ConfigRef
+//go:generate mapper method -p db -e project Exists struct=Project
+//go:generate mapper method -p db -e project Create struct=Project
+//go:generate mapper method -p db -e project UsedByRef
+//go:generate mapper method -p db -e project ID struct=Project
+//go:generate mapper method -p db -e project Rename
+//go:generate mapper method -p db -e project DeleteOne
+
+// Project represents a LXD project
+type Project struct {
+	Description string
+	Name        string   `db:"omit=update"`
+	UsedBy      []string `db:"omit=create"`
+	Config      map[string]string
+}
 
 // ProjectFilter specifies potential query parameter fields.
 type ProjectFilter struct {
-	Name string // If non-empty, return only the project with this name.
+	Name string `db:"omit=update"` // If non-empty, return only the project with this name.
```

Comment:
@masnax slightly off topic, but is there any documentation for the various field tags used by the generator that you've come across? If not is it worth starting a doc so we can refer to them? Thanks

Reply:
Right, I think documentation will definitely be necessary, especially as the complexity of the generator starts to grow.